webpack/assets/javascripts/react_app/API.js

```diff
@@ -1,37 +1,55 @@
-import $ from 'jquery';
+import axios from 'axios';
+
+const getcsrfToken = () => {
+  const token = document.querySelector('meta[name="csrf-token"]');
+
+  if (token) {
+    return token.content;
+  }
+  // fail gracefully when no token is found
+  return '';
+};
+const onSuccess = response => response.data;
+const onError = (error) => {
+  /* eslint-disable no-console */
+  console.log(error);
+  return Promise.reject(error);
+};
 
 export default {
   get(url) {
-    $.ajaxPrefilter((options, originalOptions, jqXHR) => {
-      // eslint-disable-next-line no-param-reassign
-      jqXHR.originalRequestOptions = originalOptions;
-    });
-    return $.getJSON(url);
+    return axios.get(url, {
+      headers: { 'X-Requested-With': 'XMLHttpRequest' },
+    })
+      .then(onSuccess)
+      .catch(onError);
   },
-  markNotificationAsRead(id) {
-    const data = JSON.stringify({ seen: true });
-
-    $.ajax({
-      url: `/notification_recipients/${id}`,
-      contentType: 'application/json',
-      type: 'put',
-      dataType: 'json',
-      data,
-      error(jqXHR, textStatus, errorThrown) {
-        /* eslint-disable no-console */
-        console.log(jqXHR);
+  put(url, data = {}) {
+    return axios.put(
+      url, data,
+      {
+        headers:
+      {
+        'X-Requested-With': 'XMLHttpRequest',
+        'X-CSRF-Token': getcsrfToken(),
+      },
       },
-    });
+    )
+      .then(onSuccess)
+      .catch(onError);
   },
-  markGroupNotificationAsRead(group) {
-    $.ajax({
-      url: `/notification_recipients/group/${group}`,
-      contentType: 'application/json',
-      type: 'PUT',
-      error(jqXHR, textStatus, errorThrown) {
-        /* eslint-disable no-console */
-        console.log(jqXHR);
+  post(url, data = {}, errorCallback = onError) {
+    return axios.post(
+      url, data,
+      {
+        headers:
+      {
+        'X-Requested-With': 'XMLHttpRequest',
+        'X-CSRF-Token': getcsrfToken(),
+      },
       },
-    });
+    )
+      .then(onSuccess)
+      .catch(onError);
   },
 };
```

Comment:
I think we will also need a `delete` request.

Reply:
sure we do, though right now there's no  use for it